### ISSUE TITLE
Bad syntax generated when using the Field.collate() clause in CREATE TABLE statements

### ISSUE THREAD

**Issue by db2222** [jOOQ/jOOQ#14194]
### Expected behavior

The following syntax would be ok:
```
create table [##TempTable] (
    [Random] varchar(50) collate Latin1_General_CI_AS
)
```

### Actual behavior

jOOQ instead generates the following:
```
create table [##TempTable] (
    (([Random]) collate Latin1_General_CI_AS) varchar(50)
)
```

SQL Server is unfortunately very fussy about the order of the elements...
He expects the collation to be after the data type.
Also he doesn't like the parenthesis. 

It seems like jOOQ applies the logic for select statements (https://www.jooq.org/doc/latest/manual/sql-building/column-expressions/collations/) also for the create table statement (?). But that is just my assumption.

However, in general jOOQ seems to support collate also for create table. At least according to #7118.

### Steps to reproduce the problem

```
@Test
public void sqlServerBugForCreateTableWithCollate() {
    Field<String> column = field(name("Random"), SQLDataType.VARCHAR(50));
    column = column.collate("Latin1_General_CI_AS");
    //dsl() returns a DSLContext with a configuration that uses SQLDialect.SQLSERVER2008.
    jooq.dsl().createTable(name("##TempTable")).columns(column).getSQL();
    //If execute is called instead of getSQL the exception occurs as the SQL is invalid.
}
```

### jOOQ Version

3.17.4

### Database product and version

SQL Server - Dialect 2008

### Java Version

8

### OS Version

Windows 10

**Comment by db2222** [comment]
Forgot to note. I implemented a workaround by getting the SQL from jOOQ without calling collate and then manually inserting it at the right place and running this SQL. This works. But it would be nice if manual SQL work could be avoided 😉

**Comment by lukaseder** [comment]
Thanks for your report. That's curious, given that when parsing the expected SQL, it's being re-generated correctly, e.g. online here:
https://www.jooq.org/translate

Will look into this soon

**Comment by lukaseder** [comment]
I see, there are 2 ways to attach a `Collation` to an expression:

- To `DataType::collation` (this works)
- To `Field::collate` (this doesn't work)

The intentions here are different. The former is intended to be used in DDL, the latter in queries (e.g. `ORDER BY x COLLATE y`)

But I don't think we should be so strict about these intentions. Both usages should work, because they both look as though they should work, and there's no other possible meaning of your usage than the one you intended.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.6 (https://github.com/jOOQ/jOOQ/issues/14197), and 3.16.12 (https://github.com/jOOQ/jOOQ/issues/14198).

Thanks again for your report!

**Comment by db2222** [comment]
Thank you for the very fast fix :smiley:
